Face/FindSimilar/Program.cs

```diff
@@ -0,0 +1,237 @@
+        {
+            List<string> targetImageFileNames =
+                new List<string> { "Family1-Dad1.jpg", "Family1-Daughter1.jpg", "Family1-Mom1.jpg", "Family1-Son1.jpg", "Family2-Lady1.jpg", "Family2-Man1.jpg", "Family3-Lady1.jpg", "Family3-Man1.jpg" };
+            IList<Guid?> faceIds = new List<Guid?>();
+
+            foreach (var targetImageFileName in targetImageFileNames)
+            {
+                // Read image file.
+                using (FileStream stream = new FileStream(Path.Combine("Images", targetImageFileName), FileMode.Open))
+                {
+                    // Detect faces from image stream.
+                    IList<DetectedFace> detectedFaces = Client.Face.DetectWithStreamAsync(stream).Result;
+                    if (detectedFaces == null || detectedFaces.Count == 0)
+                    {
+                        Console.WriteLine($"[Error] No face detected from image `{targetImageFileName}`.");
+                        return;
+                    }
+
+                    Console.WriteLine($"{detectedFaces.Count} faces detected from image `{targetImageFileName}`.");
+                    if (detectedFaces[0].FaceId == null)
+                    {
+                        Console.WriteLine("[Error] Parameter `returnFaceId` of `DetectWithStreamAsync` must be set to `true` (by default) for verification purpose.");
+                        return;
+                    }
+
+                    // Add detected faceId to faceIds.
+                    faceIds.Add(detectedFaces[0].FaceId);
+                }
+            }
+
+            // Find similar example for find similar faces from faceIds with similar confidence.
+            IList<SimilarFace> similarResults = Client.Face.FindSimilarAsync(faceIds[0].Value, null, null, faceIds).Result;
+            if (similarResults.Count == 0)
+            {
+                Console.WriteLine($"Not found the same with `{faceIds[0].Value}` from similar result.");
+            }
+            foreach (var similarResult in similarResults)
+            {
+                Console.WriteLine(
+                    similarResult.Confidence>0.5
+                        ? $"Faces from {faceIds[0].Value} & {similarResult.FaceId} are of the same (Positive) person, similarity confidence: {similarResult.Confidence}."
+                        : $"Faces from {faceIds[0].Value} & {similarResult.FaceId} are of different (Negative) persons, similarity confidence: {similarResult.Confidence}.");
+            }
+
+            Console.WriteLine("\nPress ENTER to exit.");
+        }
+
+        public static async Task FindSimilar_in_FaceList()
+        {
+            List<string> targetImageFileNames =
+                            new List<string> { "Family1-Dad1.jpg", "Family1-Daughter1.jpg", "Family1-Mom1.jpg", "Family1-Son1.jpg", "Family2-Lady1.jpg", "Family2-Man1.jpg", "Family3-Lady1.jpg", "Family3-Man1.jpg" };
+            string sourceImageFileName = "findsimilar.jpg";
```

Comment:
Let's take these out as class members, to be shared across functions.

Reply:
done